beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -148,9 +162,12 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 		// Get local and builder (if enabled) payloads. Set execution data. New in Bellatrix.
 		var overrideBuilder bool
 		var localPayload interfaces.ExecutionData
```

Comment:
you need this because `blobBundle` is defined outside the loop for proper closure

Reply:
I don't see how this relates to the comment, which is about localPayload.

Reply:
Because `blobsBundle` is defined in L111 outside of the if-else condition, we need to carry its lifetime

```go
	var blobBundle *enginev1.BlobsBundle
	if features.Get().BuildBlockParallel {
		blindBlobBundle, blobBundle, err = vs.BuildBlockParallel(ctx, sBlk, head, headChanged)
		if err != nil {
			return nil, errors.Wrap(err, "could not build block in parallel")
		}
	} else {
                 var overrideBuilder bool
		var localPayload interfaces.ExecutionData
		localPayload, blobBundle, overrideBuilder, err = vs.getLocalPayloadAndBlobs(ctx, sBlk.Block(), head)
		if err != nil {
			log.WithError(err).Warn("failed to retrieve local payload; attempting builder payload if available")
			if errors.Is(err, errLastMinFCU) && headChanged {
				return nil, status.Errorf(codes.Internal, "failed to retrieve local payload: %v", err)
			}
		}
          }

        fullBlobs, err = blobsBundleToSidecars(blobBundle, sBlk.Block())
		if err != nil {
			return nil, status.Errorf(codes.Internal, "Could not convert blobs bundle to sidecar: %v", err)
		}
	}

```

Reply:
The change explicitly sets 
```
		localPayload, blobBundle, overrideBuilder, err := vs.getLocalPayloadAndBlobs(ctx, sBlk.Block(), head)
```

Reply:
9a5dc4f6c